diff --git i/drivers/usb/typec/ucsi/ucsi.c w/drivers/usb/typec/ucsi/ucsi.c
index bef3f9b..a91956b 100644
--- i/drivers/usb/typec/ucsi/ucsi.c
+++ w/drivers/usb/typec/ucsi/ucsi.c
@@ -1356,7 +1356,9 @@ static void ucsi_partner_change(struct ucsi_connector *con)
 			typec_set_mode(con->port, TYPEC_MODE_AUDIO);
 			break;
 		default:
-			if (UCSI_CONSTAT(con, PARTNER_FLAG_USB))
+			/* USB and an Alternate Mode can be active together. */
+			if (UCSI_CONSTAT(con, PARTNER_FLAG_USB) &&
+			    !UCSI_CONSTAT(con, PARTNER_FLAG_ALT_MODE))
 				typec_set_mode(con->port, TYPEC_STATE_USB);
 		}
 
